src/psyclone/tests/psyad/domain/lfric/test_lfric_adjoint_harness.py

```diff
@@ -461,33 +501,58 @@ def test_generate_lfric_adj_test_quadrature(fortran_reader):
     psyir = generate_lfric_adjoint_harness(tl_psyir)
     routine = psyir.walk(Routine)[0]
     for sym in routine.symbol_table.datasymbols:
-        # All input variables should be either scalars or fields.
-        # TODO #1864 add support for operators.
+        # All input variables should be either scalars or fields (operators
+        # are never active).
         if sym.name.endswith("_input"):
             assert (sym.name.startswith("field") or
                     sym.name.startswith("rscalar"))
 
 
-def test_generate_lfric_adjoint_harness_no_operators(monkeypatch,
-                                                     fortran_reader):
-    '''Check that a kernel that has an operator as argument raises the
-    expected error. This limitation will be lifted in #1864.
+def test_generate_lfric_adjoint_harness_operator(fortran_reader,
+                                                 fortran_writer):
+    '''Check the test harness generation for a kernel that has an operator
+    as argument.
+
+    '''
+    # Alter the metadata so that the kernel expects an operator that maps
+    # from W0 to W3.
+    code = TL_CODE.replace("arg_type(gh_field,  gh_real, gh_write,  w3)  &",
```

Comment:
Hmm, that isn't very easy because in some places TL_CODE is used unmodified. Therefore I've tidied up what was there already and sorted out the inconsistent capitalisation. Is that OK?

Reply:
Fair enough.